cmd/agent/agent_test.go

```diff
@@ -281,7 +295,62 @@ func (s *testAgentSuite) testProxy(c *C) {
 	c.Assert(err, IsNil)
 }
 
-func (s *testAgentSuite) TestSimple(c *C) {
-	s.testStoreAddServer(c)
+func (s *testAgentSuite) TestProxy(c *C) {
 	s.testProxy(c)
 }
+
+func (s *testAgentSuite) checkStoreServerType(c *C, addr string, tp string) {
+	group := models.NewServerGroup(globalEnv.ProductName(), 1)
+	servers, err := group.GetServers(globalConn)
+	c.Assert(err, IsNil)
+
+	for _, server := range servers {
+		if server.Addr == addr {
+			c.Assert(server.Type, Equals, tp)
+			return
+		}
+	}
+	c.Fatalf("addr %s is not in group servers %v", addr, servers)
+}
+
+func (s *testAgentSuite) TestHA(c *C) {
+	s.agentHA = s.testStartAgent(c, "127.0.0.1:39005", true)
+	defer s.testStopAgent(c, s.agentHA)
+
+	time.Sleep(3 * time.Second)
+
+	// first test slave ha
+	procs := s.testGetProcs(c, s.agentStoreSlave)
+	// first stop agent then kill procs
+
+	s.testStopAgent(c, s.agentStoreSlave)
+	s.testKillProcs(c, procs)
+
+	time.Sleep(5 * time.Second)
```

Comment:
I don't have a good idea to determine a proper wait time for all tests like this, badly.